### PR TITLE
Game logic

### PR DESCRIPTION
**Pull request by mertcihankurel**
###Tic Tac Toe
**İn this project: **
1-) Game works without issue.
2-) Rubocop errors fixed.
3-) Players enter their name and start the game.
4-) The game continues with the loop until the game is over.

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hi Team. 👋


I understand you've given it your best. Keep going.💪

- [x] Presently there are truly issues on the board like you mentioned. It doesn't take the user input or rather doesn't seem to do anything with it.  The issue is that you haven't initialized your board. Add an object in the board initialize method that would contain the initial state of the board and then add a method in the board class that takes the player's input and inserts it on the board depending on which position the user chooses to place a mark. 

- [x] In addition, when I input a wrong value on the console I should get a warning that it's an invalid position. You would want to do this also in cases where the user inputs a position that has already been taken.

- [x] Also there shouldn't be any puts or prints a statement in the lib folder. It is meant to contain only the project logic. Kindly move all puts/prints to the bin/main.rb file.

Kindly fix these issues and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

HAPPY CODING👨‍💻

File: lib/board.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  end
+
+  def play(board)
+    turn(board) until over?(board)
+    if won?(board)
+      winner(board) == 'X' || winner(board) == 'O'
+      puts "Congratulations #{winner(board)}!"
+    elsif draw?(board)
+      puts 'Play again!'
+    end
```
Comment (mikenath223) [inline lib/board.rb]: - [x]  Move all puts/print to bin/main.rb.

File: lib/board.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+class Board
+  def initialize; end
```
Comment (mikenath223) [inline lib/board.rb]: - [x] Add an object to initialize the board in this initialize method.

File: lib/board.rb (hunk 1 above)
Comment (Haywhizzz) [reply lib/board.rb]: We have fixed everything pointed out on the review, the game now works fine

**Comment by mcihadkurel** [comment]
We changed the board. İt caused too much error.

**Review by kubilaycaglayan — CHANGES_REQUESTED** [review]
Hello Team :wave: 
Status: Changes Required :recycle: 

You can be sure that your game is working fantastically :star: 
The gameplay is flawless :+1: 

But I want to talk about the single responsibility principle with you.
**Single Responsibility** is a technical term that says every one of our methods should be responsible for only one thing. We will go over your main.rb file and look up your main loop to understand what does it mean.

loop

Here, you have a loop. The **main loop**. Which is responsible for the main game flow.
In it, you have also these things:

1. **Input Loop** : Responsible for taking inputs from the user.
2. **Input Validation**  : Responsible for validation the input. Is it a number? Is it a free spot?
3. **State Check** : Responsible for checking the state of the game after every move. Is it a win, draw or the game goes on?

What we can do here is to wrap all of those actions into isolated methods and call them one by one inside your main loop. And you can also create another method for your main loop and call it at the end.

Definitely not easy, but the real purpose of this project is to create single responsible methods and use them in each other.

As a template you can do it like this:

```ruby
def input_loop
  # input taker actions and returns the input
  # calls the input_validation at some point to make the necessary checks
end

def input_validation
  # input validating actions and returns the input or loops again
end

def state_check
  # looks at the current board and checks the state of the game and returns the state to use it in the main loop
end

def main_loop
  while game_goes_on
    # call input_loop, pass necessary parameters and return information
    # state_check, pass necessary parameters and return information
    # and as an example =>
    game_goes_on = false if state_check
  end
end

main_loop # => call for the main function to start the game
```

Surely, you need to deal with parameters and return statements. But please give it a try, then you will master **the single responsibility principle**!

- [x]  Reorganize the main loop in the ./main.rb file to make it formed in single responsible units.

Thank you very much for your time and efforts... 

With all my respect... :pray:

**Comment by Haywhizzz** [comment]
We have wrapped the actions inside isolated methods and fixed the issue

**Review by maelfosso — APPROVED** [review]
# Congratulations :clap: 

Thank you for applying the suggested changes :smiley_cat: 
You can merge it now :+1: 

Good continuation :handshake: 

Mael FOSSO

**Pull request merged**